### PR TITLE
fix: parse numeric and boolean key values

### PR DESCRIPTION
**Pull request by drktfl**
With this change, the OData request parses numerical and boolean key values into the corresponding type.

If there is an unnamed key, like in `/Entity('abc')` it is now parsed to `{'': 'abc'}` instead of the former `{abc: undefined}` which makes it easier to type it (booleans aren't valid object keys)

This should further improve the behaviour described in issue #540

### CODE REVIEW

File: packages/fe-mockserver-core/src/data/entitySets/entitySet.ts [hunk 1]
```diff
@@ -436,16 +436,8 @@ export class MockDataEntitySet implements EntitySetInterface {
...
-                        return mockData[keyName] === keyValues[keyName].substring(1, keyValues[keyName].length - 1);
-                    }
-                    return mockData[keyName] === keyValues[keyName];
                 case 'Edm.Boolean':
-                    let booleanKeyValue = keyValues[keyName];
-                    if (typeof booleanKeyValue === 'string') {
-                        booleanKeyValue = booleanKeyValue === 'true';
-                    }
-                    return mockData[keyName] === booleanKeyValue;
+                    return mockData[keyName] === keyValues[keyName];
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/data/entitySets/entitySet.ts:440]: bold !

**Review by nlunets — APPROVED** [review]
It's bold, i like it

**Review by nlunets — APPROVED** [review]
It's bold, i like it

File: packages/fe-mockserver-core/src/data/entitySets/entitySet.ts (hunk 1 above)
Comment (drktfl) [reply packages/fe-mockserver-core/src/data/entitySets/entitySet.ts:440]: if it is parsed properly...

File: packages/fe-mockserver-core/test/unit/middleware.test.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ describe('V4 Requestor', function () {
                 "/RootElement(ID=1,IsActiveEntity=true)/sap.fe.core.ActionVisibility.baseFunction(data='I am data')"
             )
             .execute('GET');
-        expect(dataRes).toMatchInlineSnapshot(`"%27I%20am%20data%27"`);
+        expect(dataRes).toMatchInlineSnapshot(`"'I am data'"`);
```
Comment (drktfl) [inline packages/fe-mockserver-core/test/unit/middleware.test.ts:105]: this is not quite right yet, it has the single quotes

Comment (nlunets) [reply packages/fe-mockserver-core/test/unit/middleware.test.ts:105]: yes almost :)

Comment (drktfl) [reply packages/fe-mockserver-core/test/unit/middleware.test.ts]: should be better now

**Review by nlunets — APPROVED** [review]

**Comment by drktfl** [comment]
Fixes #540

**Pull request merged**